pettingzoo/butterfly/pistonball/pistonball.py

```diff
@@ -544,14 +547,14 @@ def step(self, action):
                 + self.ball.velocity[0] * self.dt
             )
             if ball_next_x <= self.wall_width + 1:
-                self.done = True
+                self.terminate = True
             # ensures that the ball can't pass through the wall
             ball_min_x = max(self.wall_width, ball_min_x)
             self.draw()
             local_reward = self.get_local_reward(self.lastX, ball_min_x)
             # Opposite order due to moving right to left
             global_reward = (100 / self.distance) * (self.lastX - ball_min_x)
-            if not self.done:
+            if not self.terminate:
```

Comment:
This needs to be `if self.terminations[self.agent_selection] or self.truncations[self.agent_selection]` I think.